### PR TITLE
Add Island Portal Permission

### PR DESCRIPTION
**Pull request by Oribuin**
Added island permission for members to access portals on a player's island.

### CODE REVIEW

**Comment by Oribuin** [comment]
this is homophobia, why didn't you work

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java [hunk 1]
```diff
@@ -2,41 +2,72 @@
...
+
+        final User user = IridiumSkyblock.getInstance().getUserManager().getUser(event.getPlayer());
+        if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.PORTAL)) {
+            event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotUsePortal.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+            event.setCancelled(true);
+
+            // Teleport them back to the island to prevent constant chat spam.
+            if (event.getCause() == PlayerTeleportEvent.TeleportCause.END_PORTAL) {
+
+                if (island.get().isVisitable()) {
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:38]: what if its a private island but they are a member of the island? I should defo add an island.canVisit method that checks for island members, island trusts, island bans and island private :thinking:

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java [hunk 2]
```diff
@@ -2,41 +2,72 @@
...
+                    event.getPlayer().teleport(island.get().getCenter(islandManager.getWorld()));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:39]: Should teleport to island home not the island center

**Review by PeachesMLG — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java (hunk 2 above)
Comment (Oribuin) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:39]: aight

**Pull request merged**